internal/provider/file/resources.go

```diff
@@ -54,27 +55,48 @@ func loadFromFile(path string) (*resource.Resources, error) {
 	return resource.LoadResourcesFromYAMLBytes(bytes, false)
 }
 
-// loadFromDir loads resources from all the files under a specific directory excluding subdirectories.
 func loadFromDir(path string) ([]*resource.Resources, error) {
```

Comment:
We may need to do some work under https://github.com/envoyproxy/gateway/blob/main/internal/provider/file/file.go#L148
to make sure that any changes happened in these subdirs will also trigger the resource reload

Reply:
Let's stick to the easy way at first, can resolve any performance issues later.